### ISSUE TITLE
If using RetryPolicy server errors are not propagated to user

### ISSUE THREAD

**Issue by mmatczuk** [gocql/gocql#1161]
### What version of Cassandra are you using?

Any.

### What version of Gocql are you using?

e48272f

### What did you do?

```go
package gocql

import "testing"

func TestErrorNotPropagated(t *testing.T) {
	cluster := createCluster()
	t.Logf("Using %T", cluster.RetryPolicy)

	session, err := cluster.CreateSession()
	if err != nil {
		t.Fatal(err)
	}
	defer session.Close()

	q := session.Query("BLA BLA BLA")
	if err := q.Exec(); err != nil {
		t.Log(err)
	}
	// e48272f
	// grr_test.go:7: Using *gocql.SimpleRetryPolicy
	// grr_test.go:17: gocql: no hosts available in the pool

	// 153a728
	// grr_test.go:7: Using *gocql.SimpleRetryPolicy
	// grr_test.go:17: line 1:0 no viable alternative at input 'BLA' ([BLA]...)
}
```

### What did you expect to see?

Proper error form server:

```
line 1:0 no viable alternative at input 'BLA' ([BLA]...)
```

### What did you see instead?

Wrong information that the pool is empty

```
gocql: no hosts available in the pool
```

With the changes introduced in e48272ff, if using RetryPolicy user is not notified about server (or any other) errors.

cc @Zariel @annismckenzie @alourie

**Comment by annismckenzie** [comment]
Oh snap, I’ll take a look. 🙈😔

**Comment by annismckenzie** [comment]
Unfortunately, that's a no. I'm working on fixing it in #1162. Hopefully, it'll be done this week.

**Comment by mmatczuk** [comment]
Maybe the commit should be reverted on master?

**Comment by dahankzter** [comment]
This is a quite serious bug. Can it not be fixed in isolation @annismckenzie ?

**Comment by annismckenzie** [comment]
Yes, both of you are right – opening a PR for reverting #1164 and #1151. 😞

**Comment by annismckenzie** [comment]
PR to revert is up: #1166.

**Comment by martin-sucha** [comment]
Based on the comments above, it looks like this was fixed by the revert in #1166. Closing this issue. Please reopen if I missed something.
